src/clj/puppetlabs/puppetserver/shell_utils.clj

```diff
@@ -0,0 +1,96 @@
+(ns puppetlabs.puppetserver.shell-utils
+  (:require [schema.core :as schema]
+            [clojure.java.io :as io])
+  (:import (com.puppetlabs.puppetserver ProcessWrapper ExecutionStubImpl)
+           (org.apache.commons.exec CommandLine DefaultExecutor PumpStreamHandler)
+           (java.io IOException ByteArrayInputStream ByteArrayOutputStream)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Schemas
+
+(def ExecutionResult
+  "A map that contains the details of the result of executing a command."
+  {:exit-code schema/Int
+   :stderr schema/Str
+   :stdout schema/Str})
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Internal
+
+(schema/defn ^:private ^:always-validate
+  validate-command
+  "Checks the command string to ensure that is is an absolute path, executable
+  and that the file exists. An exception is thrown is any of those are not the
+  case."
+  [command :- schema/Str]
+  (let [command-file (io/as-file command)]
+    (cond
+      (not (.isAbsolute command-file))
+      (throw (IllegalArgumentException.
+              (format "An absolute path is required, but %s is not an absolute path" command)))
```

Comment:
Should probably throw a pair of `'` around `%s` for readability like you did below